test/libsolidity/syntaxTests/constants/constant_transient_state_variable.sol

```diff
@@ -0,0 +1,4 @@
+contract C {
+	int constant public transient x = 0;
```

Comment:
This made me think of some cases that should be covered (and work) but aren't:
```solidity
contract C {
    uint public transient pubt;
    uint internal transient it;
    uint private transient prvt;
}
```

And what about these? Looking at your current parser, I think that they will be rejected:
```solidity
contract C {
    uint transient public tpub;
    uint transient internal ti;
    uint transient private tprv;
}
```

And they should probably be accepted instead. We don't allow other locations for state variables, but looking at how the parser works, if we did they would be accepted in both positions.

This will also be important to test via a semantic test when we have codegen - a getter for a transient variable will be slightly different than one for a storage variable (`tload` vs `sload`).

Reply:
I've added the suggested tests.